### ISSUE TITLE
Data Mutation by Local Predictors

### ISSUE THREAD

**Issue by charleslparker** [bigmlcom/python#303]
This is more of a question, perhaps, than an issue, but when I passed some data through a local predictor, I was surprised to see that it was different on the other side:

```
In [22]: local = Anomaly({'object': jt, 'resource': jt['resource']})            

In [23]: instance = {'emp_length': None, 'fake_1': None, 'fake_2': 0}           

In [24]: local.anomaly_score(instance)                                          
Out[24]: 0.9481449853957093

In [25]: instance                                                               
Out[25]: {'fake_2': 0}
```

There might be a good reason for this, but it was a bit jarring for me, as I was hoping to do something with those `None` values after prediction.  So just adding a discussion here for others that may come along with the same issue, or to request a fix if this is unintentional.

**Comment by mmerce** [comment]
Not intentional at all. Internally, there's this validation that removes all fields not used in the model, but that should be done on a copy of the input data. This probably is not happening here. We should fix it.

**Comment by mmerce** [comment]
Fixed in #305
